contrib/pg_tde/src/access/pg_tde_xlog.c

```diff
@@ -107,45 +99,58 @@ tdeheap_rmgr_desc(StringInfo buf, XLogReaderState *record)
 	{
 		XLogRelKey *xlrec = (XLogRelKey *) XLogRecGetData(record);
 
-		appendStringInfo(buf, "add tde internal key for relation %u/%u", xlrec->pkInfo.data.databaseId, xlrec->mapEntry.relNumber);
+		appendStringInfo(buf, "rel: %u/%u/%u", xlrec->mapEntry.spcOid, xlrec->pkInfo.data.databaseId, xlrec->mapEntry.relNumber);
 	}
-	if (info == XLOG_TDE_ADD_PRINCIPAL_KEY)
+	else if (info == XLOG_TDE_REMOVE_RELATION_KEY)
 	{
-		TDEPrincipalKeyInfo *xlrec = (TDEPrincipalKeyInfo *) XLogRecGetData(record);
+		RelFileLocator *xlrec = (RelFileLocator *) XLogRecGetData(record);
 
-		appendStringInfo(buf, "add tde principal key for db %u", xlrec->databaseId);
+		appendStringInfo(buf, "rel: %u/%u/%u", xlrec->spcOid, xlrec->dbOid, xlrec->relNumber);
 	}
-	if (info == XLOG_TDE_EXTENSION_INSTALL_KEY)
+	else if (info == XLOG_TDE_ADD_PRINCIPAL_KEY)
 	{
-		XLogExtensionInstall *xlrec = (XLogExtensionInstall *) XLogRecGetData(record);
+		TDEPrincipalKeyInfo *xlrec = (TDEPrincipalKeyInfo *) XLogRecGetData(record);
 
-		appendStringInfo(buf, "tde extension install for db %u", xlrec->database_id);
+		appendStringInfo(buf, "db: %u", xlrec->databaseId);
 	}
-	if (info == XLOG_TDE_ROTATE_KEY)
+	else if (info == XLOG_TDE_ROTATE_PRINCIPAL_KEY)
 	{
 		XLogPrincipalKeyRotate *xlrec = (XLogPrincipalKeyRotate *) XLogRecGetData(record);
 
-		appendStringInfo(buf, "rotate principal key for %u", xlrec->databaseId);
+		appendStringInfo(buf, "db: %u", xlrec->databaseId);
 	}
-	if (info == XLOG_TDE_ADD_KEY_PROVIDER_KEY)
+	else if (info == XLOG_TDE_WRITE_KEY_PROVIDER)
 	{
 		KeyringProviderXLRecord *xlrec = (KeyringProviderXLRecord *) XLogRecGetData(record);
 
-		appendStringInfo(buf, "add key provider %s for %u", xlrec->provider.provider_name, xlrec->database_id);
+		appendStringInfo(buf, "db: %u, provider id: %d", xlrec->database_id, xlrec->provider.provider_id);
+	}
+	else if (info == XLOG_TDE_INSTALL_EXTENSION)
+	{
+		XLogExtensionInstall *xlrec = (XLogExtensionInstall *) XLogRecGetData(record);
+
+		appendStringInfo(buf, "db: %u", xlrec->database_id);
 	}
 }
 
 static const char *
 tdeheap_rmgr_identify(uint8 info)
 {
-	if ((info & ~XLR_INFO_MASK) == XLOG_TDE_ADD_RELATION_KEY)
-		return "XLOG_TDE_ADD_RELATION_KEY";
-
-	if ((info & ~XLR_INFO_MASK) == XLOG_TDE_ADD_PRINCIPAL_KEY)
-		return "XLOG_TDE_ADD_PRINCIPAL_KEY";
-
-	if ((info & ~XLR_INFO_MASK) == XLOG_TDE_EXTENSION_INSTALL_KEY)
-		return "XLOG_TDE_EXTENSION_INSTALL_KEY";
-
-	return NULL;
+	switch (info & ~XLR_INFO_MASK)
+	{
+		case XLOG_TDE_ADD_RELATION_KEY:
+			return "ADD_RELATION_KEY";
+		case XLOG_TDE_REMOVE_RELATION_KEY:
+			return "REMOVE_RELATION_KEY";
+		case XLOG_TDE_ADD_PRINCIPAL_KEY:
+			return "ADD_PRINCIPAL_KEY";
+		case XLOG_TDE_ROTATE_PRINCIPAL_KEY:
+			return "ROTATE_PRINCIPAL_KEY";
+		case XLOG_TDE_WRITE_KEY_PROVIDER:
+			return "WRITE_KEY_PROVIDER";
+		case XLOG_TDE_INSTALL_EXTENSION:
+			return "INSTALL_EXTENSION";
+		default:
+			return NULL;
```

Comment:
maybe an assert?

Reply:
I followed the pattern used by PostgreSQL.